lib/adapters/file/write.js

```diff
@@ -0,0 +1,85 @@
+var _ = require('underscore');
+var Juttle = require('../../runtime/index').Juttle;
+var fs = require('fs');
+
+var Write = Juttle.proc.sink.extend({
+    procName: 'write-file',
+    initialize: function(options, params) {
+        var allowed_options = ['file', 'limit'];
+        var unknown = _.difference(_.keys(options), allowed_options);
+        if (unknown.length > 0) {
+            throw this.compile_error('RT-UNKNOWN-OPTION-ERROR', {
+                proc: 'write file',
+                option: unknown[0]
+            });
+        }
+
+        if (!_.has(options, 'file')) {
+            throw this.compile_error('RT-MISSING-OPTION-ERROR', {
+                proc: 'write file',
+                option: "file"
+            });
+        }
+
+        this.limit = options.limit ? options.limit : 100000;
+        this.filename = options.file;
+        this.queue = [];
+    },
+
+    start: function() {
+        this.flush();
+    },
+
+    process: function(points) {
+        this.logger.debug('process', points);
+        this.queue = this.queue.concat(points);
+    },
+
+    flush: function() {
+        if (this.queue.length !== 0) {
+            var data = [];
+            var points = [];
+
+            try {
+                data = fs.readFileSync(this.filename, 'utf8');
+                points = JSON.parse(data);
```

Comment:
A thing to keep in mind (we don't have to fix it now) is that this introduces a potential race condition - other file sink / other process could do a write while we keep the points in memory, and we'd then overwrite previous contents.